File: seung365/programmers/12.py
def solution(prices):
    answer = [0] * len(prices)
    stack = []
    stack.append((prices[0],0))
    for index, price in enumerate(prices[1:-1]):
        while stack and stack[-1][0] > price:
            temp=stack.pop()
            answer[temp[1]] = (index+1) - temp[1]

        stack.append((price,index+1))

    for i in stack:
        answer[i[1]] = len(prices) - i[1] -1
    
    return answer

File: jijun0129/Algorithm/12.js
function solution(prices) {
  // stack에 index를 저장한다.
  var answer = new Array(prices.length).fill(0);
  var stack = [0];
  for (var i = 1; i < prices.length; i++) {
    //stack에 index가 저장되어 있으므로 prices를 index로 비교한다.
    while (stack.length > 0 && prices[i] < prices[stack[stack.length - 1]]) {
      // j 이후로 값이 떨어졌으므로 기간을 계산
      const j = stack.pop();
      answer[j] = i - j;
    }
    stack.push(i);
  }
  while (stack.length > 0) {
    const j = stack.pop();
    answer[j] = prices.length - 1 - j;
  }
  return answer;
}
